Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -433,6 +440,9 @@ public struct PackageBuilder {
                     if dependency.type != .library {
                         throw ModuleError.executableAsDependency(module: module.name, dependency: name)
                     }
+                    guard module.name != name else {
+                        throw ModuleError.cycleDetected("the target \(name) can't have a dependency on itself.")
+                    }
```

Comment:
No this was just for checking self dependency right now, but I guess I can just do the cycles check instead.